packages/contract/lib/utils.js

```diff
@@ -66,10 +66,14 @@ const Utils = {
         let logArgs;
         try {
           logArgs = abi.decodeLog(logABI.inputs, copy.data, copy.topics);
+          copy.args = reformat.numbers.call(
```

Comment:
I don't know if it is useful to have the rest of the params decoded, but if they are useful you could put a guard in `_convertNumber` in `contracts/lib/reformat.js` to return something benign like `""`. That way you could save the rest of the info if it can be useful later.

Reply:
Hm, maybe?  I think if something goes wrong here the results are largely going to be nonsense so I'm not sure how much it matters.  This is just a temporary workaround anyway.

Reply:
That's not really true, here the other values do get decoded going into `reformat.numbers.call`. In other words, the other values are properly decoded but there is just a problem with the data for the indexed/non-indexed value(s) not being where Truffle expects it. Again, I'm not sure if this is useful or not so this commend doesn't need to hold this up. Especially if this is going to be pull out soon.

Reply:
No but the thing is, check out the return values from `abi.decodeLog`. That is not where it is failing. All can be legit except the ones that it doesn't know where to get.

Like, for example the test function I was using had two events with the same signature but the third argument was indexed in one and not the other. The first two were decoded properly while the 3rd value was `null`.

Reply:
Yes, that what happens in *this* case, but this is a particularly nice case where the only difference is at the end.  In general, one can get nonsense values.  Like imagine what would happen if the indexing difference were in the first argument rather than the third -- you'd have a shift in the arguments, yielding nonsense values.  Or imagine that not all arguments were the same type.  Etc.  I don't think it makes sense to make an effort to report values that may be nonsense, just because it's *possible* that they're valid.

Reply:
Ok that's fine. I'm not even sure that it is that useful to decode the ones that you can haha. Looks fine to me.